prov/sockets/include/sock.h

```diff
@@ -907,12 +935,12 @@ struct sock_conn_hdr {
 	uint8_t type;
 	uint8_t reserved[3];
 	uint16_t port;
+	uint16_t data_len;
```

Comment:
sock_conn_hdr is a wire protocol structure.  We can't modify it without breaking the protocol, which requires updating the version, handling backwards compatibility, etc.  This should be dropped.  If it is needed, it must be moved to a separate patch.